staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -172,6 +173,13 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("listKind"), names.ListKind, "kind and listKind may not be the same"))
 	}
 
+	// kind cannot be a top-level type since they are shared across all groupversions
+	for _, topLevelType := range scheme.Scheme.KnownTypes(scheme.UnversionedVersion) {
+		if names.Kind == topLevelType.Name() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), names.Kind, fmt.Sprintf("kind cannot have the value: %v", topLevelType.Name())))
```

Comment:
kind %q is invalid because it is reserved

Reply:
Done.